src/Octoshift/HttpDownloadService.cs

```diff
@@ -49,5 +50,26 @@ public virtual async Task<byte[]> DownloadToBytes(string url)
 
             return await response.Content.ReadAsByteArrayAsync();
         }
+
+        public virtual async Task<Stream> DownloadStream(string url)
+        {
+            var client = new HttpClient();
+            var rnd = new Random();
+            var path = System.IO.Path.GetTempPath() + "MigrationStream" + rnd.Next(99999);
+
+            while (File.Exists(path))
+            {
+                path += rnd.Next(9); // Extra measure to prevent filename duplicity error
+            }
+
+            var response = await client.GetAsync(url, HttpCompletionOption.ResponseHeadersRead);
+
+            var streamToReadFrom = await response.Content.ReadAsStreamAsync();
+            var streamToWriteTo = File.Open(path, FileMode.Create);
+            await streamToReadFrom.CopyToAsync(streamToWriteTo);
+            File.Delete(path); // Clean up customer disc
+            return streamToWriteTo;
+
+        }
```

Comment:
there are a few things here:
1. The fact that we decided to use streams to download a file is an implementation detail so there is no need to add a new public method, we can just reimplement the already existing `DownloadToFile` method w/o touching the public interface.
2. the injected client should be used instead of instantiating a new one. 
3. I don't think we need to create a new file, we can just overwrite it. even if you want to create a new file, random is not an ideal way to create a filename since it's random not necessarily unique, instead a `Guid` should be used to guarantee uniqueness.
4. All file i/o should be abstracted out in order to be able to unit test them. We already have `FileSystemProvider` that already has the method you are looking for. 
5. `File.Delete()` should be performed in a `try-finally` block.

Reply:
Alright, I think all of this is addressed at this point too :tada:

Reply:
I'll take a final look once the PR is ready for review.